adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -442,8 +456,67 @@ private void handleConnectionRequestResult(final MqttEndpoint endpoint,
 
     private void onAuthenticationSuccess(final MqttEndpoint endpoint, final Device authenticatedDevice) {
 
+        // subscription handler for commands
+        endpoint.subscribeHandler(subscribe -> {
+            if(subscribe.topicSubscriptions().size()==1) {
+                final MqttTopicSubscription subscription = subscribe.topicSubscriptions().get(0);
+                final List<MqttQoS> grantedQosLevels = new ArrayList<>();
+                if (subscription.topicName().equals(COMMAND_REQUEST_SUBSCRIPTION)
```

Comment:
Probably `subscription.topicName()` can never be null, but I would find it more defensive to turn around the check to 
`COMMAND_REQUEST_SUBSCRIPTION.equals(subscription.topicName())`.

Reply:
This occurs at some more places in the code - just toggle the both sides of `equals`.